### PR TITLE
Auth: prevent auto_login redirect if user is already authenticated

### LINKED ISSUE #72476: Auth: auth.proxy does not work with outh and auto_login enabled
### What happened?

We have two auth plugins enabled:

```ini
[auth]
login_cookie_name = grafana_session


[auth.proxy]
enabled = true
header_name = X-WEBAUTH-USER
header_property = username
auto_sign_up = true
sync_ttl = 60
whitelist = ::1,127.0.0.1
headers = Name:X-WEBAUTH-NAME Email:X-WEBAUTH-EMAIL Groups:X-WEBAUTH-GROUPS
enable_login_token = true

[auth.azuread]
name = Azure AD
enabled = true
allow_sign_up = false
client_id = ****
client_secret = ***
scopes = openid email profile
auth_url = ***
token_url = ***
allowed_domains =
allowed_groups =
auto_login = true
```

When we make a call to `curl -k https://localhost:3000/login -H "X-WEBAUTH-USER: username"`  we do not get session cookie, but get redirected to azure auth page

### What did you expect to happen?

When auth proxy header is passed, user user should be authenticated via auth proxy, instead of being redirected to oauth

### Did this work before?

haven't tried with previous versions

### How do we reproduce it?

1. enable auth.proxy with `enable_login_token = true`
2. enable any other oauth plugin with `auto_login = true`
3. try to receive session cookie with `curl -k https://localhost:3000/login -H "X-WEBAUTH-USER: username"`

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: 10.0.1
OS: Linux
Browser: Chrome, curl

### Grafana platform?

A downloaded binary

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by mrevutskyi**
Before attemping an auto-login for oauth, verifies if current context has already been authenticated via auth-proxy

Fixes: #72476

### CODE REVIEW

File: pkg/api/login.go [hunk 1]
```diff
@@ -115,7 +115,8 @@ func (hs *HTTPServer) LoginView(c *contextmodel.ReqContext) {
 		return
 	}
 
-	if hs.tryAutoLogin(c) {
+	// If user is not already signed in via auth-proxy, try auto-login
```
Comment (kalleep) [inline pkg/api/login.go:118]:
```suggestion
	// If user is not authenticated try auto-login
```

**Review by kalleep — APPROVED** [review]
Looks good. I think instead of adding this check the if c.IsSignedIn block below could be moved to also prevent this from happening and skipping some additional logic from happening. But I would need to test that

**Comment by mrevutskyi** [comment]
Thank you. Accepted your suggestions.

Not sure why `clone-enterprise` check is failing https://drone.grafana.net/grafana/grafana/129534/1/2

**Comment by kalleep** [comment]
I saw that you said that the email used is the primary for your github account but for some reason github does not link it correctly

**Comment by mrevutskyi** [comment]
Figured it out, I  put my email into git's `user.name` instead of `git.email` 🤦‍♂️ 
Fixed now

**Comment by kalleep** [comment]
@mrevutskyi Thanks for your contribution

**Pull request merged**